examples/gno.land/p/sys/vals/poc/poc.gno

```diff
@@ -0,0 +1,93 @@
+package poc
+
+import (
+	"std"
+
+	"gno.land/p/sys/vals/types"
+)
+
+// PoC specifies the Proof of Contribution validator set.
+// In order to become part of the set, users to be voted into
+// the validator set by a govdao proposal
+type PoC struct {
+	// validators holds the current validator set.
+	// This slice can never practically grow more than ~150 elements,
+	// due to Tendermint's quadratic network complexity
+	validators              []*types.Validator
+	addressToValidatorIndex map[std.Address]int // address -> index
+}
+
+// NewPoC creates a new empty Proof of Contribution validator set
+func NewPoC(opts ...Option) *PoC {
+	// Create the empty set
+	p := &PoC{
+		validators:              make([]*types.Validator, 0),
+		addressToValidatorIndex: make(map[std.Address]int),
+	}
+
+	// Apply the options
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	return p
+}
+
+func (p *PoC) AddValidator(address std.Address, pubKey string) *types.Validator {
+	// Validate that the operation is a valid call
+	p.validateAdd(address)
+
+	v := &types.Validator{
+		Address:     address,
+		PubKey:      pubKey, // TODO: in the future, verify the public key
+		VotingPower: 1,      // in this PoC system, all new validators have the same voting power
```

Comment:
@moul

Do you think we should have the voting power be part of the AddValidator API?

Reply:
yes

Reply:
Added in:
[d81d300](https://github.com/gnolang/gno/pull/2130/commits/d81d300a5e2ae5b4fc598a691bfc6ff10319a1f3)